crates/subspace-fraud-proof/src/tests.rs

```diff
@@ -127,13 +127,7 @@ async fn execution_proof_creation_and_verification_should_work() {
     .await;
 
     // Bob is able to sync blocks.
-    futures::join!(
-        alice.wait_for_blocks(1),
-        bob.wait_for_blocks(1),
-        ferdie.produce_blocks(1),
-    )
-    .2
-    .unwrap();
+    produce_blocks!(ferdie, alice, bob, 1).await.unwrap();
```

Comment:
I have mixed feelings about this. We had a clear distinction between, who is producing and who is waiting. Not feels like everyone is producing blocks. Not a Blocker though

Reply:
The domain nodes are also producing blocks, but they are subject to the primary node instead of producing blocks proactively, thus we are using `wait_for_blocks` internally.